src/widget/wstarrating.cpp

```diff
@@ -37,55 +36,27 @@ void WStarRating::setup(const QDomNode& node, const SkinContext& context) {
 QSize WStarRating::sizeHint() const {
     QStyleOption option;
     option.initFrom(this);
-    QSize widgetSize = style()->sizeFromContents(QStyle::CT_PushButton, &option,
-                                                 m_starRating.sizeHint(), this);
+    QSize widgetSize = style()->sizeFromContents(QStyle::CT_PushButton,
+            &option,
+            m_visualStarRating.sizeHint(),
+            this);
 
     m_contentRect.setRect(
-        (widgetSize.width() - m_starRating.sizeHint().width()) / 2,
-        (widgetSize.height() - m_starRating.sizeHint().height()) / 2,
-        m_starRating.sizeHint().width(),
-        m_starRating.sizeHint().height()
-    );
+            (widgetSize.width() - m_visualStarRating.sizeHint().width()) / 2,
+            (widgetSize.height() - m_visualStarRating.sizeHint().height()) / 2,
+            m_visualStarRating.sizeHint().width(),
+            m_visualStarRating.sizeHint().height());
 
     return widgetSize;
 }
 
-void WStarRating::slotTrackLoaded(TrackPointer pTrack) {
-    if (m_pCurrentTrack != pTrack) {
-        if (m_pCurrentTrack) {
-            disconnect(m_pCurrentTrack.get(), nullptr, this, nullptr);
-            m_pCurrentTrack.reset();
-        }
-        if (pTrack) {
-            connect(pTrack.get(),
-                    &Track::changed,
-                    this,
-                    &WStarRating::slotTrackChanged);
-            m_pCurrentTrack = pTrack;
-        }
-        updateRatingFromTrack();
-    }
-}
-
-void WStarRating::setRating(int rating) {
-    // Check consistency with the connected track
-    DEBUG_ASSERT(!m_pCurrentTrack ||
-            m_pCurrentTrack->getRating() == rating);
-    m_starRating.setStarCount(rating);
-    update();
-}
-
-void WStarRating::updateRatingFromTrack() {
-    if (m_pCurrentTrack) {
-        setRating(m_pCurrentTrack->getRating());
-    } else {
-        setRating(0);
+void WStarRating::slotSetRating(int starCount) {
+    if (starCount == m_starCount) {
+        // Unchanged
+        return;
     }
-}
-
-void WStarRating::slotTrackChanged(TrackId trackId) {
-    Q_UNUSED(trackId);
-    updateRatingFromTrack();
+    m_starCount = starCount;
+    updateVisualRating(starCount);
```

Comment:
had to add `emit ratingChanged(m_starCount)` in order to make the _up/_down controls to also update the track value
pushed a fix 0550e07eddcb7d45256038c67fac13f40d52328d